cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureResourceConnector.java

```diff
@@ -133,7 +133,7 @@
         boolean resourcesPersisted = false;
         try {
             List<CloudResource> instances;
-            if (!client.templateDeploymentExists(resourceGroupName, stackName)) {
+            if (createTemplateDeployment(stackName, resourceGroupName, client)) {
```

Comment:
minor: I would change the name to
`shouldCreateTemplateDeplyoment` or to  `isTemplateDeploymentNeeded`
to reflect that this is a method deciding what to do and it will not create the template.

But given that my reveiw is late, I do not insist on this.